scripts/tag_release.sh

```diff
@@ -0,0 +1,199 @@
+        Tag the release even if VERSION is equal to or older than the most
+        recent release. If a tag already exists for VERSION, delete it.
+EOF
+}
+
+################################################################################
+# Handle Arguments
+################################################################################
+
+FORCE=
+
+parse_args() {
+    # Process command line arguments.
+    args=()
+    while [ "$1" != "" ]; do
+    case $1 in
+        -h | --help)
+            show_usage
+            exit 0
+            ;;
+        -f | --force)
+            FORCE=T
+            ;;
+        --)
+            shift
+            args+=($*)
+            break
+            ;;
+        *)
+            if [[ "$1" == -* ]]; then
+              args+=($1)
+            else
+              args+=("$1")
+            fi
+            ;;
+    esac
+    shift
+    done
+}
+
+parse_args "$@"
+set -- "${args[@]}"
+
+VERSION=$1
+if [[ -z "${VERSION}" ]]; then
+  printf "Error: Version not specified.\n\n"
+  show_usage
+  exit 1
+fi
+
+COMMIT=$2
+if [[ -z "${COMMIT}" ]]; then
+  COMMIT="HEAD"
+fi
+
+TAG="v${VERSION}"
+
+################################################################################
+# Run Application
+################################################################################
+
+pushd ${REPO_ROOT} >/dev/null
+
+# Check if the specified version is already tagged.
+echo "Checking requested version (${VERSION})."
+if git describe --tags --abbrev=0 --match ${TAG} >/dev/null 2>/dev/null; then
+  if [[ -n "${FORCE}" ]]; then
+    echo "A tag with the specified version already exists. Deleting and overwriting."
+    git tag -d ${TAG}
+  else
+    echo "Error: A tag with the specified version already exists. Use --force to overwrite."
+    exit 2
+  fi
+fi
+
+# Checkout the requested commit.
+if [[ "${COMMIT}" != "HEAD" ]]; then
+  echo "Checking out requested commit."
+  git checkout ${COMMIT} 2>/dev/null
+fi
+
+COMMIT=$(git rev-parse HEAD)
+COMMIT_MESSAGE=$(git log --format=%s -n1 HEAD)
+echo "Using commit ${COMMIT} (${COMMIT_MESSAGE})."
+
+# Get the most recently released version _before_ the requested commit. If the
+# requested commit is older than the latest release on current master, this
+# will not be the latest release version. This is useful if you need to create
+# a patch to a previous release, but current master is on a newer major/minor
+# version. For example, if the latest release is 1.15.0, you might still want
+# to release version 1.14.1 to patch a bug in 1.14.0.
+PREV_VERSION=$(git describe --tags --abbrev=0 --match "v*" | sed -e 's/^v//')
+PREV_COMMIT=$(git rev-parse v${PREV_VERSION})
+echo "Found previous version ${PREV_VERSION} (commit ${PREV_COMMIT})."
+
+# Check if the new version is older than the current version.
+cat <<EOF | python3
+from packaging import version
+import sys
+
+prev_version = version.parse("${PREV_VERSION}")
+new_version = version.parse("${VERSION}")
+
+if new_version <= prev_version:
```

Comment:
You should add some error message to explain why the script exited here. Something along the lines of:
"This branch was previously tagged with ${PREV_VERSION} which is a newer version number then the specified tag ${VERSION}"